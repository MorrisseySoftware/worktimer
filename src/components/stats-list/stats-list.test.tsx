import React from 'react'
import { act, create, ReactTestRenderer } from 'react-test-renderer'
import { shallow, mount, render } from 'enzyme'
import StatsList from './stats-list'
import {
    TimeLeft,
    TimerCompletionState,
} from 'components/timer-display/timer-display'

describe('Stats List Tests', () => {
    it('should update the display list when the history prop changes', () => {
        const historyItems = [
            {
                timer: {
                    minutes: 15,
                    seconds: 27,
                    milliseconds: 75,
                } as TimeLeft,
                completionTime: new Date(2019, 1, 1, 7, 11, 0, 0),
                completed: false,
            } as TimerCompletionState,
        ]
        let component: ReactTestRenderer = {} as ReactTestRenderer
        act(() => {
            component = create(
                <StatsList history={[]} deleteCallBack={() => {}} />
            )
        })
        expect(component.toJSON()).toMatchSnapshot()
        act(() => {
            component.update(
                <StatsList history={historyItems} deleteCallBack={() => {}} />
            )
        })
        expect(component.toJSON()).toMatchSnapshot()
    })
    // it('should toggle the button state from OFF to ON', () => {
    //     let component: ReactTestRenderer = {} as ReactTestRenderer
    //     act(() => {
    //         component = create(<StatsList history={[]} />)
    //     })
    //     expect(component.toJSON()).toMatchSnapshot()
    //     act(() => {
    //         component.update(<StatsList history={[]} />)
    //     })
    //     expect(component.toJSON()).toMatchSnapshot()
    // })
    // it('should call a click handler when the toggle button is clicked', () => {
    //     const component = shallow(<StatsList history={[]} />)
    //     component.find('button').simulate('click')
    //     expect([]).toBe('called!')
    // })
})
