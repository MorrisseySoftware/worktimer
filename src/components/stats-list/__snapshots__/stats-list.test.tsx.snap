// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats List Tests should update the display list when the history prop changes 1`] = `
<div
  className="history__container"
>
  <div
    className="history__list"
  >
    NO TIMERS
  </div>
  <div
    className="history__total"
  >
    Total Time - 
    None
  </div>
</div>
`;

exports[`Stats List Tests should update the display list when the history prop changes 2`] = `
<div
  className="history__container"
>
  <div
    className="history__list"
  >
    <p
      className="stat__container"
    >
      <span
        className="stat__label"
      >
        07:11 AM - 15:27:75
      </span>
      <button
        className="delete__btn"
        onClick={[Function]}
      >
        Delete
      </button>
    </p>
  </div>
  <div
    className="history__total"
  >
    Total Time - 
    15:27:75
  </div>
</div>
`;
